Added constants file, removing constants from php/utils.php

SaraMcCutcheon opened the pull request:
Adding the get-your-refresh-token URL to the utils file was heinous (It took 8 lines!) and so inspired me to get the default constants moved out of the code.

Commit:
Added constants file, removing constants from php/utils.php

bensheldon commented:
Can we embed the environment variables dynamically into the YAML so we just have a single "constants" object that defines all of the sources?

http://symfony.com/legacy/doc/reference/1_2/en/02-yaml#chapter_02_dynamic_yaml_files

bensheldon commented:
Just to expand, I don't like the idea of having a constants file that we then dynamically parse and munge with other variables.

... though looking at it more, I see there is a very clear 1-to-1 mapping between how the ENV variables are constructed and constants. If we added an explanation to the top of the `constants.yml` file that explained how these can be modified via ENV (and where to go look for that code), that would satisfy me.

SaraMcCutcheon commented:
It doesn't seem that Symfony has the ability to process inline PHP in 3.x, and in my attempt all PHP added was passed through as a string. I looked through the code and couldn't find any way to make it discern inline PHP, and so suppose that the capability was lost somewhere between 1.2 (as linked) and 3.0. 

I added the note at the top of the constants file and changed the names of the vars to match the constants' names for clarity.

As you are un-fond of this idea, do you have other ideas to better manage the default constant values? It has become a bit of a mess and is due to get worse [once Auth0 login is implemented](https://github.com/pantheon-systems/cli/pull/790/files#diff-e02dc22fa7c2dd0a3059084ee686ec10R139).

bensheldon commented:
This looks good. Thanks for digging through the interpolation; I guess it's a lot cleaner in Ruby's yml parser (which I'm familiar with) to interpolate values

I'm glad you were able to pull them out into a single location. Thank you!
